source/modules/MEI.shared.xml

```diff
@@ -1855,7 +1855,7 @@
     <attList>
       <attDef ident="meiversion" usage="opt">
         <desc>Specifies a generic MEI version label.</desc>
-        <defaultVal>4.0.1</defaultVal>
+        <defaultVal>4.0.1-dev</defaultVal>
```

Comment:
Can we have a default value that is not in the closed list? This does not make sense to me.

Reply:
yeah you're right, missed that

Reply:
Since 4.0.1-dev is already out it should be 4.1.0-dev or 5.0.0-dev, shoudn't it?

Reply:
@lpugin is right, but we'll always have that problem – our `dev`-version will always be named in relation to something that comes after the thing we know the name of. The only solution I see at this point is to add all potential successors to the list of allowed values, so basically something like
```xml
<valList type="closed">
  <valItem ident="4.0.2-dev">
    <desc>Potential bugfix release 4.0.2</desc>
  </valItem>
  <valItem ident="4.1.0-dev">
    <desc>Potential minor release 4.1.0</desc>
  </valItem>
  <valItem ident="5.0.0-dev">
    <desc>Potential first release of MEI 5</desc>
  </valItem>
</valList>
```
While we work on the schema, we can escalate those levels – as long as we have only bugfixes, our `defaultVal` will be `4.0.2-dev`. When we have sufficiently complex changes, we will take out the 4.0.2 and 4.1.0 options one after another. When we do a release, we just pick the smallest value still available. It seems arguable to keep the 4.0.0 and 4.0.1 releases in the list, as long as we don't break compatibility. At the same time, I like to be strict about the dev version… Thoughts?

Reply:
Regarding the question of whether we do a `4.1.0` or a `5.0.0` now – I'm still in favor of having a clean 5 now. Yes, the changes aren't dramatic, but there are some things which aren't allowed anymore, beyond bugfixes. The only thing that let's me hesitate is that we don't have the PDF yet, and I would prefer to have that together with the next major release. We could delay the release by a couple of weeks now, or we do a 4.1 now, where I feel less urged to have the PDF in place. However, this is dangerously close to http://sentimentalversioning.org/, and I certainly don't want to get there. Actually, I like the idea of an automated comparison with the previous release to decide about the severity of changes. I did that with the tool that generated https://music-encoding.org/archive/comparison-4.0.html, and it spotted a small number of breaking changes. In order to avoid sentiments in this, I think that even a single such change on an element that's barely ever used has to trigger a new major release. Comments?

Reply:
I don't think potential bug fixes should be included in the develop branch. What we currently have in develop should never become 4.0.2. If we make a bug fix, we take master and then integrate the bugfix from master into develop.